### ISSUE TITLE
Warning allennlp/nn/util.py:149 using Pytorch 1.0

### ISSUE THREAD

**Issue by TheShadow29** [allenai/allennlp#1878]
**Describe the bug**
Not a bug but a warning. 
```
allennlp/nn/util.py:149: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than tensor.new_tensor(sourceTensor).
  index_range = sequence_lengths.new_tensor(torch.arange(0, len(sequence_lengths)))
```

**To Reproduce**
I installed allennlp via pip. Removed existing pytorch (since it was 0.4) and installed pytorch-nightly. I am following the elmo tutorial (https://github.com/allenai/allennlp/blob/master/tutorials/how_to/elmo.md#using-elmo-as-a-pytorch-module-to-train-a-new-model). After the embeddings line I am getting the warning mentioned above. 
Attaching a screenshot for reference
![image](https://user-images.githubusercontent.com/10602903/46588103-50b1fd00-ca4b-11e8-9a96-d19c21bf6ef7.png)

**Expected behavior**
There should be no warning

**System (please complete the following information):**
 - OS: Ubuntu 16.04
 - Python version: 3.7
 - AllenNLP version: v0.7.0
 - PyTorch version: pytorch-nightly ('1.0.0.dev20181007')

**Additional context**
Reference file in the repo: https://github.com/allenai/allennlp/blob/master/allennlp/nn/util.py#L149

**Comment by matt-gardner** [comment]
We should be able to fix this using something like `.to(sequence_lengths.device())`.  Just FYI for anyone who starts working on this.

**Comment by schmmd** [comment]
We're going to upgrade to PyTorch 1.0 when it's fully released (expected in early December).  If there's any fixes compatible with our current version we'll take contributions!

**Comment by schmmd** [comment]
Closing a @joelgrus fixed these when migrating to PyTorch 1.0
